test/models/track_test.rb

```diff
@@ -176,6 +176,58 @@ class TrackTest < ActiveSupport::TestCase
     assert_nil Track.for_repo("exercism/configlet")
   end
 
+  test "" do
+    track = create :track, slug: 'ruby'
+    assert_equal track, Track.for_repo("exercism/#{track.slug}")
+  end
+
+  test ".id_for_repo with track repo with repo url" do
+    track = create :track, repo_url: "https://github.com/exercism/ruby"
+    assert_equal track.id, Track.id_for_repo(track.repo_url)
+  end
+
+  test ".id_for_repo with track repo with repo name" do
+    track = create :track, repo_url: "https://github.com/exercism/ruby"
+    assert_equal track.id, Track.id_for_repo("exercism/ruby")
+  end
+
+  %w[test-runner analyzer representer].each do |suffix|
+    test ".id_for_repo with #{suffix} repo with repo url" do
+      track = create :track, repo_url: "https://github.com/exercism/ruby"
+      assert_equal track.id, Track.id_for_repo("#{track.repo_url}-#{suffix}")
+    end
+
+    test ".id_for_repo with #{suffix} repo with repo name" do
+      track = create :track, repo_url: "https://github.com/exercism/ruby"
+      assert_equal track.id, Track.id_for_repo("exercism/ruby-#{suffix}")
+    end
+  end
+
+  [
+    ["exercism/dotnet-tests", "csharp"],
+    ["exercism/nim-docker-base", "nim"],
+    ["exercism/codemirror-lang-arturo", "arturo"],
+    ["exercism/codemirror-lang-phix", "phix"],
+    ["exercism/codemirror-lang-gleam", "gleam"],
+    ["exercism/codemirror-lang-wren", "wren"],
+    ["exercism/codemirror-lang-elixir", "elixir"],
+    ["exercism/highlightjs-arturo", "arturo"],
+    ["exercism/highlightjs-gdscript", "gdscript"],
+    ["eslint-config-typescript", "typescript"],
+    ["eslint-config-javascript", "javascript"],
+    ["eslint-config-tooling", "javascript"],
+    ["javascript-lib-static-analysis", "javascript"]
```

Comment:
```suggestion
    ["exercism/babel-preset-javascript", "javascript"],
    ["exercism/babel-preset-typescript", "typescript"],
    ["exercism/javascript-lib-jest-extensions", "javascript"],
    ["exercism/dotnet-tests", "csharp"],
    ["exercism/nim-docker-base", "nim"],
    ["exercism/codemirror-lang-arturo", "arturo"],
    ["exercism/codemirror-lang-phix", "phix"],
    ["exercism/codemirror-lang-gleam", "gleam"],
    ["exercism/codemirror-lang-wren", "wren"],
    ["exercism/codemirror-lang-elixir", "elixir"],
    ["exercism/highlightjs-arturo", "arturo"],
    ["exercism/highlightjs-gdscript", "gdscript"],
    ["eslint-config-typescript", "typescript"],
    ["eslint-config-javascript", "javascript"],
    ["eslint-config-tooling", "typescript"],
    ["javascript-lib-static-analysis", "javascript"]
```

tooling is both, static analysis is both, so this evens that out.